python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -316,6 +320,74 @@ cpdef read_parquet(filepaths_or_buffers, columns=None, row_groups=None,
         df._data.label_dtype = cudf.dtype(column_index_type)
     return df
 
+cpdef read_parquet_metadata(filepaths_or_buffers):
+    """
+    Cython function to call into libcudf API, see `read_parquet_metadata`.
+
+    filters, if not None, should be an Expression that evaluates to a
+    boolean predicate as a function of columns being read.
+
+    See Also
+    --------
+    cudf.io.parquet.read_parquet
+    cudf.io.parquet.to_parquet
+    """
+    # Convert NativeFile buffers to NativeFileDatasource,
+    for i, datasource in enumerate(filepaths_or_buffers):
+        if isinstance(datasource, NativeFile):
+            filepaths_or_buffers[i] = NativeFileDatasource(datasource)
+
+    cdef cudf_io_types.source_info source = make_source_info(filepaths_or_buffers)
+
+    args = move(source)
+
+    cdef parquet_metadata c_result
+
+    # Read Parquet metadata
+    with nogil:
+        c_result = move(parquet_metadata_reader(args))
```

Comment:
To explicitly use move assignment similar to `read_parquet` and elsewhere in the file to keep things consistent.

Reply:
I'll let it slide, but would still like to learn more about this.
@galipremsagar does this move have any effect? In C++ it would not be required because `parquet_metadata_reader(args)` is a temporary. I'd love to understand if this works in the same way in Cython.

Reply:
> In C++ it would not be required because parquet_metadata_reader(args) is a temporary.

My thoughts exactly as move assignment would automatically be used in C++ in such scenario

Reply:
The tl;dr is yes, the move is necessary. It has to do with the way Cython-generated code looks. I wrote up an explanation of this for someone else once, let me see if I can find it.

Reply:
https://github.com/rapidsai/cudf/pull/9571#discussion_r740424808

Reply:
Thanks @vyasr !